src/Frontend/Core.hs

```diff
@@ -0,0 +1,16 @@
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE OverloadedStrings   #-}
+
+{-# OPTIONS_GHC -Werror -Wall #-}
+
+module Frontend.Core
+where
+
+import Data.Typeable
+import Data.String.Conversions
+import Lucid
+import Lucid.Base
+
+
+semanticDiv :: forall m a. (Monad m, ToHtml a, Typeable a) => a -> HtmlT m () -> HtmlT m ()
+semanticDiv t = div_ [makeAttribute "data-aula-type" (cs . show . typeOf $ t)]
```

Comment:
This will generate the following snippet:
````html
<div data-aula="PageIdea"> ... </div>
````

Which serves two purposes:
 * It helps the front-en developer to identify which part of the generated pages comes from which combinator
 * Later on when we write selenium suite, the semantic tags helps up to parse, identify and test elements on the page.